internal/runtime/network.go

```diff
@@ -160,3 +167,25 @@ func pubsubTopic(match string) func(string) bool {
 		return match == strings.TrimPrefix(s, prefix)
 	}
 }
+
+type Advertiser struct {
```

Comment:
Nice - I've been meaning to do this.

A small thing:  let's not export this type.  Rename to `advertiserService`?